File: rev-com-server/src/main/java/com/codergb/bean/Movie.java
package com.codergb.bean;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Movie  extends FileInfo{
  private String id;
  private String name;
  private List<Director> director;
  private List<Screenwriter> screenwriter;
  private String coverUrl;
  private String area;
  private String language;
  private Date releaseTime;
  private Long duration;
  private String alias;
}

File: rev-com-manage/src/types/movie.ts
import { IDirector } from "@/types/director"
import { IActor } from "@/types/actor"
import { ICategory } from "@/types/category"
import { IScreenwriter } from "@/types/screenwriter"
import { IArea } from "@/types/area"
import { IForm } from "@/types/form"

interface IMovie {
  id: string
  name: string
  directors: IDirector[]
  screenwriters: IScreenwriter
  coverUrl: string
  area: IArea
  language: string
  releaseTime: string
  duration: number
  alias: string
  form: IForm
  actors: IActor[]
  categories: ICategory
  description: string
}
export type { IMovie }
